tests/unit_tests/test_analytics/test_sentry.py

```diff
@@ -0,0 +1,387 @@
+import os
+import socket
+import threading
+import time
+import zlib
+
+import flask
+import pytest
+import requests
+import wandb
+import wandb.analytics
+import wandb.env
+from flask import request
+from sentry_sdk.envelope import Envelope
+from sentry_sdk.utils import sentry_sdk
+
+EXTERNAL_SENTRY_DSN_KEY = "EXTERNALKEY"
+EXTERNAL_SENTRY_PROJECT = "123456"
+INTERNAL_SENTRY_DSN_KEY = "INTERNALKEY"
+INTERNAL_SENTRY_PROJECT = "654321"
+CLIENT_MESSAGE = "client message"
+WANDB_MESSAGE = "wandb message"
+API_SERVER_ADDRESS = "127.0.0.1:{port}"
+SENTRY_DSN_FORMAT = "http://{key}@{address}/{project}"
+WANDB_DSN_URL_FORMAT = SENTRY_DSN_FORMAT.format(
+    key=INTERNAL_SENTRY_DSN_KEY,
+    address=API_SERVER_ADDRESS,
+    project=INTERNAL_SENTRY_PROJECT,
+)
+EXTERNAL_DSN_URL_FORMAT = SENTRY_DSN_FORMAT.format(
+    key=EXTERNAL_SENTRY_DSN_KEY,
+    address=API_SERVER_ADDRESS,
+    project=EXTERNAL_SENTRY_PROJECT,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def relay():
+    relay_server = MetricRelayServer()
+    relay_server.start()
+
+    return relay_server
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_env(relay):
+    os.environ[wandb.env.ERROR_REPORTING] = "True"
+    os.environ[wandb.env.SENTRY_DSN] = WANDB_DSN_URL_FORMAT.format(port=relay.port)
```

Comment:
i would think it is better to just patch it directly in the tests, this is quite a big side-effect of turning the error reporting on